src/utils/histogram.cc

```diff
@@ -0,0 +1,129 @@
+/*
+ This file is part of Leela Chess Zero.
+ Copyright (C) 2018 The LCZero Authors
+
+ Leela Chess is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Leela Chess is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "utils/histogram.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <algorithm>
+#include <cmath>
+
+Histogram::Histogram()
+    : Histogram(kDefaultMinExp, kDefaultMaxExp, kDefaultMinorScales) {}
+
+Histogram::Histogram(int minExp, int maxExp, int minorScales)
+    : minExp_(minExp),
```

Comment:
According to the style guide, variable names should be like `min_exp` and member variable names like `min_exp_`